app/controllers/devise_registration_controller.rb

```diff
@@ -180,6 +180,8 @@ def create
         user_is_confirmed: false,
         user_is_new: true,
       }
+
+      record_security_event(SecurityActivity::USER_CREATED, user: resource)
```

Comment:
I ended up putting this here... still wondering about this a bit...

The [record_security_event](https://github.com/alphagov/govuk-account-manager-prototype/blob/97d5d55f5dbde15c85ea8f13db7dd69b93365ec5/app/controllers/application_controller.rb#L13-L21) method also pulls in the `request` from which it gets the user agent and IP to help users see if there's anything suspicious to report...

My other plan had been to use an [after_create](https://github.com/rails/rails/blob/914caca2d31bd753f47f9168f2a375921d9e91cc/activerecord/lib/active_record/callbacks.rb#L372) active record callback. That firing in conjunction with the model being created feels more solid to me somehow compared to an action in a controller.

But in the end the inclusion of request data steered me over here... and I think the only benefit (aside from neatness) to doing it on the model would be if we looked at validations to guarentee this event existed. Not sure if that's a good idea anyway.

Up for other thoughts?

Reply:
I think doing it here makes sense.